Datastore/composer.json

```diff
@@ -5,6 +5,7 @@
     "minimum-stability": "stable",
     "require": {
         "google/cloud-core": "^1.43",
+        "google/auth": "^1.18",
```

Comment:
We shouldn't need to require this, it is already resolved by `google/cloud-core` :)

Reply:
@dwsupplee seems like I will have to keep the version to `"google/cloud-firestore": "dev-main"` else it fails to do composer update in System tests setup, [link to relevant workflow file](https://github.com/googleapis/google-cloud-php/blob/main/.github/workflows/datastore-emulator-system-tests.yaml#L29-L31):

```
google/cloud-firestore v1.25.3 requires google/cloud-core ^1.39 
-> satisfiable by google/cloud-core[v1.39.0, ..., v1.47.2] from composer repo (https://repo.packagist.org) 

but google/cloud-core[dev-feat_datastore_multiple_db, dev-main] 
from path repo (../Core) has higher repository priority. 

The packages from the higher priority repository do not match your constraint and are therefore not installable. 
That repository is canonical so the lower priority repo's packages are not installable. 
See https://getcomposer.org/repoprio for details and assistance
```

CC: @saranshdhingra

Reply:
Would we be able to follow a similar pattern as used for core in the workflow for firestore?